### ISSUE TITLE
Incoming webhook failed because of certificate issues

### ISSUE THREAD

**Issue by Subhasish5** [SparebankenVest/akv2k8s-website#73]
Internal error occurred: failed calling webhook
        "pods.env-injector.admission.spv.no": failed to call webhook: Post
        "https://akv2k8s-envinjector.akv2k8s.svc:443/pods?timeout=10s": x509:
        certificate has expired or is not yet valid: current time

How to fix this issue

**Comment by Subhasish5** [comment]
Please note that we deployed the latest version of helmrelease in our cluster and while trying to scale up our application pod, we found the below error :

I0801 13:53:51.622048 1 main.go:139] "found pod to mutate" pod="otxecm/"
I0801 13:53:51.622077 1 pod.go:286] "creating client certificate to use with auth service" otxecm/="(MISSING)"
I0801 13:53:51.635961 1 pod.go:294] "create authentication service secret" otxecm/="(MISSING)"
I0801 13:53:51.676808 1 pod.go:308] "mutate init-containers" otxecm/="(MISSING)"
I0801 13:53:51.676827 1 pod.go:314] "mutate containers" otxecm/="(MISSING)"
I0801 13:53:51.676834 1 pod.go:126] "found container to mutate" container="otxecm/dmsapi"
I0801 13:53:51.676839 1 pod.go:129] "checking for env vars to inject" container="otxecm/dmsapi"
I0801 13:53:51.676848 1 pod.go:132] "found env var to inject" env="dmsapi-otclient-password@azurekeyvault" container="otxecm/dmsapi"
E0801 13:53:51.712734 1 main.go:162] "failed to mutate" err="failed to get auto cmd, error: cannot fetch image descriptor: GET https://econcrdev.azurecr.io/oauth2/token?scope=repository%3Admsapi%3Apull&service=econcrdev.azurecr.io: UNAUTHORIZED: authentication required, visit https://aka.ms/acr/authorization for more information." pod="otxecm/"
2023/08/01 13:53:51 [ERROR] admission webhook error: failed to get auto cmd, error: cannot fetch image descriptor: GET https://econcrdev.azurecr.io/oauth2/token?scope=repository%3Admsapi%3Apull&service=econcrdev.azurecr.io: UNAUTHORIZED: authentication required, visit https://aka.ms/acr/authorization for more information.
2023/08/01 14:00:10 http: TLS handshake error from 10.244.8.34:43062: remote error: tls: bad certificate


Could you please help us how to renew the akv2k8s certificate.

Thanks and Regards,
Subhasish

**Comment by tspearconquest** [comment]
Did your certificate expire?

`kubectl get secret akv2k8s-envinjector-tls -n akv -o jsonpath='{.data.tls\.crt}' | base64 -d | openssl x509 -noout -text`

**Comment by Subhasish5** [comment]
@tspearconquest  yes the certificate has been expired...can you please let me know how do we renew this certificate as well
